### ISSUE TITLE
[BUG] Spliced Transform

### ISSUE THREAD

**Issue by MetallurgyBlockworks** [pagefaultgames/pokerogue#2481]
**Describe the bug**
A lock-up occurs if a spliced Ditto transforms into another Pokemon during a double battle. The battle continues normally, but once the Dittosplice gains experience it locks up.

This bug has been replicated on two different floors transforming into four different Pokemon during double battles.

**To Reproduce**
1. Splice a Shiny 1 Ditto with a Spidops
2. Enter a double battle
3. Transform into either opposing Pokemon
4. Win the battle

**Expected behavior**
The Ditto gains the experience and life moves on.

**Screenshots / Videos**
![image](https://github.com/pagefaultgames/pokerogue/assets/162029849/9946683e-cb7d-4cd3-89b6-3447dbd85d31)

![image](https://github.com/pagefaultgames/pokerogue/assets/162029849/55f3a642-0f0f-4ed3-ba0d-25a34cb57cf4)

![image](https://github.com/pagefaultgames/pokerogue/assets/162029849/815091cc-ae2a-4d6d-9af1-f1331681db45)

![image](https://github.com/pagefaultgames/pokerogue/assets/162029849/27e60215-66f9-4093-951e-61cc238a1a1e)

![image](https://github.com/pagefaultgames/pokerogue/assets/162029849/e33fc8ee-48b8-482a-8f2a-5ef3b7ce942a) (Game is frozen at this screen.)

**Comment by SivWatt** [comment]
Bumped into this bug splicing shiny Ditto with another Pokemon (Solrock #338). 
This will always happen when winning any fight with transformed spliced Ditto in court. 
For me it happened in classic wave 200.

**Comment by Snailman11** [comment]
Grand Transform Aggregated Issues (Fusion Transform Softlock, Delayed Visual, Form Not Saved, Stats not copied(imposter), pp not reduced)
https://discord.com/channels/1125469663833370665/1267976989957029900/1267976989957029900

**Comment by LoaneShark** [comment]
Can confirm that it is still happening, in my case I spliced an imposter ditto into the second slot (first slot is a shiny furret with pickup passive). It was also during a classic mode run.

To be more specific, the game softlocks if any of these 3 things happen happen:
1) You check the summary of the ditto fusion while it is transformed (and on the field)
2) You win a battle / try to change areas while the ditto fusion is transformed (and therefore on the field)
3) You try to teach it a move or encounter an event that checks your moves while it is transformed

Basically anything that opens the summary screen or tries to check the movelist of the ditto fusion while it is transformed into something else. My furretto happened to have all 4 move slots filled, in this case.
